### PR TITLE
Upgrade to nwb 0.11.0 & fixes #39 (concat vendor/app scripts)

### LINKED ISSUE #39: At build time, concat vendor/app scripts and async load in
Basically, over in this: https://github.com/insin/react-hn/blob/master/src/views/index.ejs#L37

We want to concatenate both vendor and app scripts into a single bundle so at build time instead of:

``` html
    <script src="build/vendor.js"></script>
    <script src="build/app.js"></script>
```

We're just doing:

``` html
    <script src="build/bundle.js" async></script>
```

to avoid blocking.

An alternative to this approach would be ignoring concatenation and just focusing on our HTTP/2 story. I haven't deployed HTTP/2 with push support on AppEngine before, but could learn how to do so if we wanted it.

### PR DESCRIPTION
**Pull request by addyosmani**
This patchset upgrades us to the latest version of `nwb` and also moves us over to async loading in the single bundle generated by `vendorBundle: false`.

### CODE REVIEW

**Comment by insin** [comment]
LGTM

I want to check if there's a difference between what gets bundled using the `react` command vs. using the `nwb` command, as they have different presets. The `react` command was intended for standalone serving of an entry point sitting anywhere, but looks like it works just fine for projects too :+1:

**Comment by addyosmani** [comment]
SGTM :) Thanks for taking a look! The one difference I did notice was what appeared to be gzipped files being output to the project root directory. I don't believe I changed anything else here that would cause that but I was unsure if that was an intended default now or something we could turn off (or a bug) :)

**Comment by NekR** [comment]
Seems fine, but I wasn't able to figure out how `nwb` works, so no idea how it affects build. If @insin says it's okay, then it has to be okay :-)

**Comment by insin** [comment]
Er, yeah, that sounds weird, I'll check it out locally tonight.

**Comment by addyosmani** [comment]
@insin By any chance, were you able to repro what I was seeing? :) I'd be happy to switch back to the `nwb` command to see if those artefacts still get output with latest.

**Comment by addyosmani** [comment]
Once we've had a chance to investigate further, I'll update so https://github.com/insin/react-hn/issues/39 is tackled. Not super high priority atm.

**Comment by addyosmani** [comment]
@insin friendly ping just in case you get a chance to look at this :) If you're busy I can take a look at moving back to `nwb` over `react`

**Comment by insin** [comment]
@addyosmani thanks, on my radar again

Is there any way to get GitHub's [Assigned Issues](https://github.com/issues/assigned) view to not include `is:issue` by default? I've started to appreciate the focus you can get by assigning stuff to yourself and only ever looking at that when you have some time to spare.

**Comment by insin** [comment]
I'm not seeing the gzipped file thing, but there are a couple of other issues I hit which I can add a commit to this branch for

**Comment by addyosmani** [comment]
@insin I think you might be able to use a bookmark to get the desired filtering there. For example [this](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+assignee%3Ainsin) gives you the view you were after I believe. If I misunderstood, holler 📯 

Thanks for the extra commits and tweaks to this branch! Looks like it requires a little rebasing against master. Will  take a look at that.

File: package.json [hunk 1]
```diff
@@ -9,37 +9,39 @@
...
-    "lint:fix": "./node_modules/eslint-config-jonnybuchanan/bin/lint.js --fix .",
+    "lint": "eslint src server.js",
+    "lint:fix": "npm run lint -- --fix",
     "start": "node server.js",
     "postinstall": "npm run build",
-    "serve": "./node_modules/.bin/nwb serve",
-    "precache": "./node_modules/sw-precache/cli.js --root=public --config=sw-precache-config.json"
+    "serve": "nwb serve",
+    "precache": "sw-precache --root=public --config=sw-precache-config.json"
   },
```
Comment (addyosmani) [inline package.json]: I think I may have needed to use these paths to get the Node on GAE deploys happy again. I'll try it out with these changes and see if we run into any trouble at all.

**Comment by insin** [comment]
I genuinely forgot that bookmarks are a thing :+1:

**Comment by addyosmani** [comment]
Haha. Okay, soooo..updated this to a merge-friendly state. Local builds are all good.

I am still seeing those mysterious .json.gzip files in my builds, even with latest `nwb` installed. If however you aren't able to repro, I'll assume I may need to tinker with my `nwb` versions a little to see what's causing this. 

screen shot 2016-07-16 at 09 21 07

Otherwise looks good to go.

**Comment by insin** [comment]
It looks like those are caused by the `cacheDirectory` setting in babel-loader, which nwb turns on with `true` config:

https://github.com/webpack/webpack/issues/2223

They go into the Local Temp directory for me on Windows, wonder why you're getting them in the current dir.

**Comment by addyosmani** [comment]
Cheers for the info! Is there a way we could explicitly set `cacheDirectory` via `nwb` and see if that changes things on Mac? I do think there's probably a weird OS-specific bug here that be more of an upstream problem.

**Comment by insin** [comment]
Are you running as root? Searching around it seems like that results in the cwd being the temp dir.

**Comment by addyosmani** [comment]
Ah. That may well be it. I'm working on a Goog corp machine with some extra root weirdness applied to `npm` to work with our setup. If it's a known issue upstream there I can resolve this locally and we should be good to merge these changes.

**Comment by addyosmani** [comment]
This PR has been replaced with the work that just landed in master updating us to more recent versions of nwb and deps as well as adding in route-based code-splitting. Closing this partic PR off as it's outdated now.

**Pull request closed without merging**